hunting/okta/docs/queries/credential_access_rapid_reset_password_requests_for_different_users.toml

```diff
@@ -0,0 +1,32 @@
+[hunt]
+author = "Elastic"
+description = """
+This hunting query identifies rapid reset password requests for different users in Okta. Adversaries may attempt to reset passwords for multiple users in rapid succession to gain unauthorized access to accounts or disrupt operations. This query identifies when the source user is different from the target user in reset password events and filters for users with more than 15 reset password attempts.
+"""
+integration = ["okta"]
+uuid = "c784106e-6ae8-11ef-919d-f661ea17fbcc"
+name = "Rapid Reset Password Requests for Different Users"
+language = ["ES|QL"]
+license = "Elastic License v2"
+notes = [
+    "`okta.actor.alternate_id` is the potentially compromised account",
+    "An API access token may have been compromised, where okta.actor.alternate_id reflects the owner",
+    "To identify a list of tokens this user created, search for the `okta.actor.alternate_id` where `event.action` is `system.api_token*` which may require a larger time window"
+]
+mitre = ['T1098.001']
+query = ['''
+from logs-okta.system*
+| where @timestamp > NOW() - 7 day
+
+// Filter for reset password events where the source user is different from the target user
+| where event.dataset == "okta.system" and event.action == "user.account.reset_password" and source.user.full_name != user.target.full_name
+
+// Extract relevant fields
+| keep @timestamp, okta.actor.alternate_id, okta.debug_context.debug_data.dt_hash, user.target.full_name, okta.outcome.result
+
+// Count the number of reset password attempts for each user
+| stats reset_attempts = count(*) by okta.actor.alternate_id, user.target.full_name, okta.debug_context.debug_data.dt_hash
```

Comment:
If I understood correctly the logic is to detect  `Password Requests for Different Users` but the current `stats` aggregate by `user.target.full_name` shouldn't be this `source.user.full_name` ? (same source user resetting pwd for different accounts can be expressed) : 

`| stats reset_attempts = count(*),  unique_count_target_user = count_distinct(user.target.full_name) by source.user.full_name
| where  unique_count_target_user >= 15 or reset_attempts >= 15`

Reply:
Correct, this works much better, fixed!